lib/src/widgets/content_item/poll.dart

```diff
@@ -0,0 +1,140 @@
+        children: [
+          ..._choices.map(
+            (choice) => Stack(
+              alignment: AlignmentGeometry.centerLeft,
+              children: [
+                if (_showResults)
+                  ConstrainedBox(
+                    constraints: BoxConstraints(
+                      maxHeight:
+                          2 *
+                          (Theme.of(context).textTheme.bodyLarge?.fontSize ??
+                              20),
+                    ),
+                    child: FractionallySizedBox(
+                      heightFactor: 1,
+                      widthFactor: max(
+                        choice.numVotes / max(totalVotes, 1),
+                        0.00001,
+                      ),
+                      child: DecoratedBox(
+                        decoration: BoxDecoration(
+                          color: Theme.of(
+                            context,
+                          ).colorScheme.primary.withValues(alpha: 0.5),
+                          borderRadius: BorderRadius.circular(15),
+                        ),
+                      ),
+                    ),
+                  ),
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                  children: [
+                    Row(
+                      children: [
+                        if (widget.poll.multiple)
+                          Checkbox(
+                            value: choice.chosen,
+                            onChanged: (newValue) async {
+                              final post = await ac.api.threads.votePoll(
+                                widget.poll.postId,
+                                choice.id,
+                              );
+                              setState(() {
+                                _choices =
+                                    post.poll?.choices.toList() ?? _choices;
+                              });
+                            },
+                          ),
+                        if (!widget.poll.multiple)
+                          Radio<int>(
+                            value: choice.id,
+                            enabled: _selectedAnswer == null,
+                          ),
+                        Text(choice.text),
+                      ],
+                    ),
+                    if (_showResults)
+                      Text(
+                        '${((choice.numVotes / max(totalVotes, 1)) * 100).toInt()}%',
+                      ),
+                  ],
+                ),
+              ],
+            ),
+          ),
+          Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              Text('$totalVotes votes'),
+              ElevatedButton(
+                onPressed: () => setState(() {
+                  _showResults = !_showResults;
+                }),
+                child: Text('Show results'),
```

Comment:
This should be localized, and should be either "Show results" or "Hide results" based on the `_showResults` variable.